src/api/api.ts

```diff
@@ -7,141 +14,200 @@ import {
   IUser,
   IWorkflow,
   IWrikeResponse,
-  RequestMethod,
+  IAccount,
+  RequestParams,
 } from "./types";
+import type { Settings } from "../types";
 
 export const getCustomFields = async (
-  client: IDeskproClient
+  client: IDeskproClient,
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<ICustomFields[]>> =>
-  installedRequest(client, `api/v4/customfields`, "GET");
+  request(client, { endpoint: `api/v4/customfields`, method: "GET", settings });
 
 export const createNote = async (
   client: IDeskproClient,
   taskId: string,
-  comment: string
+  comment: string,
+  settings: RequestParams["settings"],
 ) =>
-  installedRequest(client, `api/v4/tasks/${taskId}/comments`, "POST", {
-    text: comment,
+  request(client, {
+    endpoint: `api/v4/tasks/${taskId}/comments`,
+    method: "POST",
+    data: { text: comment },
+    settings,
   });
 
 export const getNotesByTaskId = async (
   client: IDeskproClient,
-  taskId: string
-) => installedRequest(client, `api/v4/tasks/${taskId}/comments`, "GET");
+  taskId: string,
+  settings: RequestParams["settings"],
+) => request(client, {
+  endpoint: `api/v4/tasks/${taskId}/comments`,
+  method: "GET",
+  settings,
+});
 
 export const editTask = async (
   client: IDeskproClient,
   taskId: string,
-  data: unknown
-) => installedRequest(client, `api/v4/tasks/${taskId}`, "PUT", data);
+  data: unknown,
```

Comment:
Is this because the content is dynamic? I'd assume it's still an object so can we type that?

Reply:
I haven't dealt with the types in this app yet. However, I created a story where I'll review and fix the types
https://app.shortcut.com/deskpro/story/161047/wrike-improvements